gradle-nonnull-plugin/plugin/src/main/groovy/com.novoda.gradle.nonnull/GeneratePackageAnnotationsTask.groovy

```diff
@@ -0,0 +1,71 @@
+package com.novoda.gradle.nonnull
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileVisitDetails
+import org.gradle.api.tasks.TaskAction
+
+class GeneratePackageAnnotationsTask extends DefaultTask {
+
+    def outputDir
+
+    def variant
+
+    @TaskAction
+    void generatePackageAnnotations() {
+        description = "Annotates the source packages with @ParametersAreNonnullByDefault"
+
+        variant.sourceSets.each {
```

Comment:
I'm not sure if supported, but using `.all` instead of `.each` when possible would be more robust I guess. Probably worth checking this option.